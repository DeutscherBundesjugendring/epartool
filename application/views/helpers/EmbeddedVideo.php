<?php

/**
 * Class Admin_View_Helper_EmbeddedVideo
 */
class Application_View_Helper_EmbeddedVideo extends Zend_View_Helper_Abstract
{

    /**
     * @return string
     */
    public function embeddedVideo($service, $id)
    {
        if (method_exists($this, $service)) {
            return $this->$service($id);
        }
        
        throw new Exception('Service ' . $service . ' for embedding video is not defined.');
    }
    
    /**
     * @param string $id
     * @return string
     */
    private function facebook($id)
    {
        return '<div class="embed-responsive embed-responsive-16by9">
                <div
                    class="fb-video"
                    data-href="https://www.facebook.com/facebook/videos/' . $id . '/"
                    data-width="520"
                    data-show-text="false"
                ></div>
            </div>';
    }
    
    /**
     * @param string $id
     * @return string
     */
    private function vimeo($id)
    {
        return '<div class="embed-responsive embed-responsive-16by9">
                <iframe
                    src="https://player.vimeo.com/video/' . $id . '?title=0&byline=0&portrait=0&badge=0"
                    width="640"
                    height="480"
                    frameborder="0"
                    webkitallowfullscreen
                    mozallowfullscreen
                    allowfullscreen
                ></iframe>
            </div>';
    }
    
    /**
     * @param string $id
     * @return string
     */
    private function youtube($id)
    {
        return '<div class="embed-responsive embed-responsive-16by9">
                <iframe
                    id="ytplayer"
                    type="text/html"
                    width="640"
                    height="480"
                    src="https://www.youtube.com/embed/' . $id . '?autoplay=0"
                    frameborder="0"
                ></iframe>
            </div>';
    }
}
