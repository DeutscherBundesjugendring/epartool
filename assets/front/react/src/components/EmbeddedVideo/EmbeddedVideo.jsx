import React from 'react';

/* global embedVideoUrl */

const EmbeddedVideo = (props) => {
  let src = null;

  if (embedVideoUrl[props.videoService] !== undefined) {
    src = embedVideoUrl[props.videoService].replace('%s', props.videoId);
  }

  if (src) {
    return (
      <iframe
        width={props.width}
        height={props.height}
        src={src}
        className="followup-video"
      />
    );
  }

  return null;
};

EmbeddedVideo.defaultProps = {
  width: 240,
  height: 120,
};

EmbeddedVideo.propTypes = {
  videoService: React.PropTypes.oneOf(['youtube', 'vimeo', 'facebook']).isRequired,
  videoId: React.PropTypes.string.isRequired,
  width: React.PropTypes.number,
  height: React.PropTypes.number,
};

export default EmbeddedVideo;
