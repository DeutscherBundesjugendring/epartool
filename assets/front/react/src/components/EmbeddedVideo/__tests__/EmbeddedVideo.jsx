import { shallow } from 'enzyme';
import React from 'react';
import EmbeddedVideo from '../EmbeddedVideo';

global.embedVideoUrl = {
  facebook: 'https://www.facebook.com/video/embed?video_id=%s',
  vimeo: 'https://player.vimeo.com/video/%s',
  youtube: 'https://www.youtube.com/embed/%s',
};

describe('rendering', () => {
  it('renders correctly youtube embedded video', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="youtube"
        videoId="youtube-video-id"
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly youtube embedded video with specific size', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="youtube"
        videoId="youtube-video-id"
        width={1920}
        height={1080}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly vimeo embedded video', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="vimeo"
        videoId="vimeo-video-id"
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly vimeo embedded video with specific size', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="vimeo"
        videoId="vimeo-video-id"
        width={1920}
        height={1080}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly facebook embedded video', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="facebook"
        videoId="facebook-video-id"
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly facebook embedded video with specific size', () => {
    const tree = shallow(
      <EmbeddedVideo
        videoService="facebook"
        videoId="facebook-video-id"
        width={1920}
        height={1080}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
