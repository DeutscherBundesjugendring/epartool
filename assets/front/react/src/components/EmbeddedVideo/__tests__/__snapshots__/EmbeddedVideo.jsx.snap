// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly facebook embedded video 1`] = `
<iframe
  className="followup-video"
  height={120}
  src="https://www.facebook.com/video/embed?video_id=facebook-video-id"
  title="Video facebook-video-id"
  width={240}
/>
`;

exports[`rendering renders correctly facebook embedded video with specific size 1`] = `
<iframe
  className="followup-video"
  height={1080}
  src="https://www.facebook.com/video/embed?video_id=facebook-video-id"
  title="Video facebook-video-id"
  width={1920}
/>
`;

exports[`rendering renders correctly vimeo embedded video 1`] = `
<iframe
  className="followup-video"
  height={120}
  src="https://player.vimeo.com/video/vimeo-video-id"
  title="Video vimeo-video-id"
  width={240}
/>
`;

exports[`rendering renders correctly vimeo embedded video with specific size 1`] = `
<iframe
  className="followup-video"
  height={1080}
  src="https://player.vimeo.com/video/vimeo-video-id"
  title="Video vimeo-video-id"
  width={1920}
/>
`;

exports[`rendering renders correctly youtube embedded video 1`] = `
<iframe
  className="followup-video"
  height={120}
  src="https://www.youtube.com/embed/youtube-video-id"
  title="Video youtube-video-id"
  width={240}
/>
`;

exports[`rendering renders correctly youtube embedded video with specific size 1`] = `
<iframe
  className="followup-video"
  height={1080}
  src="https://www.youtube.com/embed/youtube-video-id"
  title="Video youtube-video-id"
  width={1920}
/>
`;
